clippy_lints/src/collapsible_if.rs

```diff
@@ -163,6 +164,54 @@ impl CollapsibleIf {
         }
     }
 
+    fn check_collapsible_if_nested_if_else(context: &LateContext<'_>, if_expr: &Expr<'_>) {
+        let ExprKind::If(cond, then, else_opt) = if_expr.kind else {
+            return;
+        };
+
+        let ExprKind::Block(then_block, _) = then.kind else {
+            return;
+        };
+
+        let Some(then_expr) = then_block.expr else {
+            return;
+        };
+
+        let ExprKind::If(inner_cond, inner_then, _) = then_expr.kind else {
+            return;
+        };
+
+        let Some(else_expr) = else_opt else {
+            return;
+        };
+
+        let mut spanless_eq = SpanlessEq::new(context);
+
+        if !spanless_eq.eq_expr(inner_then, else_expr) {
+            return;
+        }
+
+        span_lint_and_then(
+            context,
+            COLLAPSIBLE_IF,
+            if_expr.span,
+            "this `if` statement can be collapsed.",
+            |diag| {
+                diag.span_suggestion(
+                    if_expr.span,
+                    "collapse else and nested if blocks",
+                    format!(
+                        "if !{} || {} {}",
+                        cond.span.get_source_text(context).unwrap().to_owned(),
+                        inner_cond.span.get_source_text(context).unwrap().to_owned(),
+                        else_expr.span.get_source_text(context).unwrap().to_owned()
```

Comment:
You should use the `Sugg` or `snippet` interface with proper applicability checking.